### DISCUSSION TITLE
Unable to trigger automation

### DISCUSSION THREAD

**Discussion by rj-budibase** [Budibase/budibase#13990] [Help]
Hi
I am new to budibase, hence using the free version. I tried to create an automation, when i create a new row in table1, a new row has to be created in table2. When i tried the test run, I found that my table2 gets the new row as expected but the first table1(For which Row Created trigger has to work) , no row is getting created . How can i trouble shoot this.

**Comment by mikesealey** [comment]
Hey @rj-budibase I can see you've also asked this in the discord server, but in case you don't see it there, I'll copy my response over to you here as well.

When you're working on your automation you should use the "Finish and Test Auomation button", which will give you the opportunity to supply some test data - in your case it will be a chance to mock up a row in the first table to trigger a row being created in the second table. 

Take a closer look at the inputs and outputs of the automation steps on the right - it might help you understand what's going wrong.
![image](https://github.com/Budibase/budibase/assets/39269380/7ecf9857-2187-403b-a006-8bb7b56fa3c1)

The entire section of "Automation" in the documentation makes for great reading, it's a really complex and powerful area of Budibase, so it's well worth a look - https://docs.budibase.com/docs/automation-steps

**Reply by rj-budibase** [reply]
Hi Mike ,
Thank you for your response. I checked thoroughly  the inputs and outputs of my test automation steps and here is it. I am unable to find a mistake in this. Could you please help me find it out.Row Created

"id": "",
"row": {
"tableId": "datasource_plus_cd33dc10f596453f99486de9d832dd9d__table1",
"COL1": "USA",
"COL2": "Canada"
},
"automation": {
"_id": "au_2a60913bf7804cff8c280b187889287e",
"_rev": "50-ea9f5336f52c71fc3d326381c76d1cbf",
"name": "testautomation",
"type": "automation",
"definition": {
"steps": [
{
"name": "Create Row",
"tagline": "Create a {{inputs.enriched.table.name}} row",
"icon": "TableRowAddBottom",
"description": "Add a row to your database",
"type": "ACTION",
"internal": true,
"features": {
"LOOPING": true
},
"stepId": "CREATE_ROW",
"inputs": {
"schema": {...},
"row": {
"tableId": "datasource_plus_cd33dc10f596453f99486de9d832dd9d__table2",
"tab2col1": "Apricot",
"tab2col2": "Banana"
}
},
"schema": {
"inputs": {
"properties": {...},
"required": [...]
},
"outputs": {
"properties": {...},
"required": [...]
}
},
"id": "SahW0BVMG"
}
],
"trigger": {...}
},
"disabled": false,
"appId": "app_dev_skillveritrainingsolutionspvtltd_e42a975651124bafb38d05a30f56dbdb",
"createdAt": "2024-06-25T07:24:25.266Z",
"updatedAt": "2024-06-25T09:29:47.509Z",
"testData": {
"id": "",
"row": {
"tableId": "datasource_plus_cd33dc10f596453f99486de9d832dd9d__table1",
"COL1": "USA",
"COL2": "Canada"
}
}
}
}

Create Row
{
"row": {
"tableId": "datasource_plus_cd33dc10f596453f99486de9d832dd9d__table2",
"tab2col1": "Apricot",
"tab2col2": "Banana"
},
"response": {
"id": 29,
"tab2col1": "Apricot",
"tab2col2": "Banana",
"_id": "%5B29%5D",
"tableId": "datasource_plus_cd33dc10f596453f99486de9d832dd9d__table2",
"_rev": "rev"
},
"id": "%5B29%5D",
"revision": "rev",
"success": true

**Comment by PClmnt** [comment]
@rj-budibase Table1 won't be updated by the automation in this case. The automation you have created will only update Table2 when a new row is created in Table1. i.e in the design section you have a form that saves a new row to Table1, when this happens it will perform the Create Row step. You can think of the first step (the trigger) as a listener. It waits to see if something happens in Table1, and if it does then performs the rest of the steps. Hopefully this makes some sense!

**Reply by rj-budibase** [reply]
Hi Peter ,
Thank you for the response. I understood what you meant.  I am using the free version of Budibase and in that I have found at https://account.budibase.app/portal/upgrade that I can use 20 + triggers and run 200 automations per month. At the same time in the automation creation page it says trigger automation is a permium feature.  It is confusing . Though my test automation was successful, when i tried to run the form, the automation didnt work. Does that mean I can trigger it only if I buy the premium version? If so the listing of automation facility under the free account section is totally confusing. Could you please clarify this
